plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -446,14 +448,22 @@ private static long getTargetSnapshotIdFromVersion(Table table, ConnectorTableVe
         return snapshotId;
     }
 
-    private static long getTemporalSnapshotIdFromVersion(Table table, ConnectorTableVersion version, io.trino.spi.type.Type versionType)
+    private static long getTemporalSnapshotIdFromVersion(ConnectorSession session, Table table, ConnectorTableVersion version, io.trino.spi.type.Type versionType)
     {
         if (versionType instanceof TimestampWithTimeZoneType timeZonedVersionType) {
             long epochMillis = timeZonedVersionType.isShort()
                     ? unpackMillisUtc((long) version.getVersion())
                     : ((LongTimestampWithTimeZone) version.getVersion()).getEpochMillis();
             return getSnapshotIdAsOfTime(table, epochMillis);
         }
+        else if (versionType instanceof DateType) {
+            long epochMillis = LocalDate.ofEpochDay((Long) version.getVersion())
+                    .atStartOfDay()
```

Comment:
When the snapshot time is 2023-07-26 01:02:03 and SELECT query specifies `FOR TIMESTAMP AS OF DATE '2023-07-26`, the snapshot wouldn't be included, right?

Does this behavior conform to SQL standard? Could you leave the code comment?